### DISCUSSION TITLE
Abstract generics with new() contraint

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [dotnet/csharplang#6180] [Language Ideas] [answered]
What do you guys think of this?
```cs
public abstract class A
{

}

public class B : A
{

}

public class Example<T> where T : A, new() // new currently not possible with abstracts
{
    T a;

     public Example()
     {
          a = new T();
     }
}

var test1 = new Example<A>(); // error!
var test2 = new Example<B>(); // success!
```

I run into this issue a lot and I feel that I shouldn't need to sacrifice my abstract methods to use generics to construct a object like this.
The way I see it is the new constraint should only make sure the type passed has a parameterless constructor.

There is also no issue with not knowing the type because we pass the type B directly in. so Example<T> knows it's creating a instance of type B

**Comment by chrisoverzero** [comment, marked as answer]
What is the question or suggestion here? The code sample [appears to work](https://sharplab.io/#v2:CYLg1APg+gBAvDAdgUwO4wKIA8CGBbABwBtkAeAQQD4AKASgG4YB6JmZAJ3YHt2BCAWABQsBCnTZ8xMgCEaDZqwDOAVwDGq5IsUDBQgAIBmGDgBGigC7scq8zD0AmGOSEBvIQF8h+ow5jShMIEwIE6uHl6ChnaOEoQkpAAqlAFBqAAWHMgwCcFOADRIaHQKheiqypzIiOZEAJ5IXLYEXFoAliYkMKit5mnGZpbW5ophgkHZTjAuMADmyOaM7vCldPThgkA==) as you have commented.

**Reply by Shadowblitz16** [reply]
Interesting this is .net 6?

**Reply by Shadowblitz16** [reply]
I'm dumb sorry for wasting your time. I guess generics get confusing when your using them alot. thank goodness they aren't c++ templates.

My issue was the fact that my parameterless constructors aren't implemented automatically when I derive from my base class.

**Reply by chrisoverzero** [reply]
> […] sorry for wasting your time.

No worries at all. Everyone is always learning.
